src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,19 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
+if PY3 and pathlib is not None:  # pragma: no branch
+    _global_type_lookup[pathlib.Path] = \
```

Comment:
> Not convinced that cross-OS support is a good idea, but have added PurePath

Fair enough.  In that case though, it would still be good to resolve to the native concrete path type - if there's a strategy registered for WindowsPath it can also be used for Path, PureWindowsPath, and PurePath as it's a subtype of all those things.  Registering strategies for the parent types is strictly optional (but nice to have).  Eg:

```
path_strings = st.fspaths(allow_pathlike=False).map(os.fsdecode)
if os.name == 'nt':
    os_paths = [pathlib.WindowsPath, pathlib.PureWindowsPath]
else:
    os_paths = [pathlib.PosixPath, pathlib.PurePosixPath]
_global_type_lookup.update({
    t: path_strings.map(t) for t in os_paths + [pathlib.Path, pathlib.PurePath]
})
```